packages/paypal-commerce-integration/src/paypal-commerce-ratepay/paypal-commerce-ratepay-payment-strategy.ts

```diff
@@ -331,13 +331,20 @@ export default class PaypalCommerceRatepayPaymentStrategy implements PaymentStra
             );
 
             const isOrderApproved = orderStatus === PayPalOrderStatus.PollingStop;
+            const isPollingError = orderStatus === PayPalOrderStatus.PollingError;
 
             if (isOrderApproved) {
                 this.deinitializePollingMechanism();
 
                 return resolvePromise();
             }
 
+            if (isPollingError) {
+                this.deinitializePollingMechanism();
+
+                return rejectPromise();
+            }
```

Comment:
Should we throw an error in case if polling was failed?

Reply:
Can you cover new changes with tests?

Reply:
In which case polling error status returns from backend?

Reply:
Discussed in DM: This status can be received from BE if PayPal has some issues on their side. So to avoid infinite polling loop it will be stopped